src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -31,16 +33,38 @@ private List<object> Buffer
                     _buffer = new List<object>();
                 }
 
+                if (_isSingleContentSet)
+                {
+                    Debug.Assert(_buffer.Count == 0);
+
+                    _buffer.Add(_singleContent);
+                    _isSingleContentSet = false;
+                }
```

Comment:
These are pooled. If you reinitialize a `TagHelperContent` the buffer may not be `null`. I could make the changes to always use the normal buffer once it's initialized but that liters the code with a lot more `if (_buffer != null)` checks. Didn't seem worth it for the same functionality.